src/pages/workspaces/workspace/workflows/WorkflowView.js

```diff
@@ -86,36 +86,46 @@ const ioTask = ({ name }) => _.nth(-2, name.split('.'))
 const ioVariable = ({ name }) => _.nth(-1, name.split('.'))
 const ioType = ({ inputType, outputType }) => (inputType || outputType).match(/(.*?)\??$/)[1] // unify, and strip off trailing '?'
 
+
 const WorkflowIOTable = ({ which, inputsOutputs: data, config, errors, onChange, onSetDefaults, onBrowse, suggestions, readOnly }) => {
+  const [sort, setSort] = useState({ field: 'taskVariable', direction: 'asc' })
+
+  const sortedData = _.orderBy([function(o) {
+    return Utils.switchCase(sort.field,
+      ['taskVariable', () => ioTask(o).toLowerCase()],
+      ['workflowVariable', () => ioVariable(o).toLowerCase()]
+    )
+  }, 'optional'], [sort.direction, 'asc'], data)
```

Comment:
Should 'optional' be first? I assume the goal is to put all optionals at the bottom (when not hidden).

Reply:
Oh right okay. So you're thinking it should go all items without optionals alphabetized then all items with optionals alphabetized?

Reply:
That feels right to me